src/bokeh/models/widgets/tables.py

```diff
@@ -858,6 +859,55 @@ def __init__(self, *args, **kwargs) -> None:
     The height of each row in pixels.
     """)
 
+    @staticmethod
+    def from_data(data, columns=None, formatters={}, **kwargs) -> DataTable:
+        """ Create a simple table from a pandas dataframe, dictionary or ColumnDataSource.
+
+        Args:
+            data (DataFrame or dict or ColumnDataSource) :
+                The data to create the table from. If the data is a dataframe
+                or dictionary, a ColumnDataSource will be created from it.
+
+            columns (list, optional) :
+                A list of column names to use from the input data.
+                If None, use all columns. (default: None)
+
+            formatters (dict, optional) :
+                A mapping of column names and corresponding Formatters to
+                apply to each column. (default: None)
+
+        Keyword arguments:
+            Any additional keyword arguments will be passed to DataTable.
+
+        Returns:
+            DataTable
+
+        Raises:
+            ValueError
+                If the provided data is not a ColumnDataSource
+                or a data source that a ColumnDataSource can be created from.
+
+        """
+
+        if isinstance(data, ColumnDataSource):
+            data = data
```

Comment:
`data = data` will be a head scratcher for anyone reading this code later. I think just `source = data.clone()` here `source = ColumnDataSource(data)` in the next branch would be less surprising (and also avoid the unnecessary clone in case the input is not a CDS to begin with)

Reply:
Just to clarify a bit here, are you saying that if the initial input data is *not* a CDS, then we don't need to clone it?

Reply:
Correct, that branch is already creating a brand new CDS here in side the funcion, so we are free to modify it in the function without cloning first. The thing we want to avoid is modifying a CDS that the user passed in.